src/Uno.UI.Runtime.Skia.X11/X11XamlRootHost.cs

```diff
@@ -309,23 +312,62 @@ private void Initialize()
 		}
 
 		IntPtr window;
+		uint depth = 32;
 		if (FeatureConfiguration.Rendering.UseOpenGLOnX11 ?? IsOpenGLSupported(display))
 		{
 			_x11Window = CreateGLXWindow(display, screen, size);
 			window = _x11Window.Value.Window;
 		}
 		else
 		{
-			window = XLib.XCreateSimpleWindow(
-				display,
-				XLib.XRootWindow(display, screen),
-				0,
-				0,
-				(int)size.Width,
-				(int)size.Height,
-				0,
-				XLib.XBlackPixel(display, screen),
-				XLib.XWhitePixel(display, screen));
+			var matchVisualInfoResult = XLib.XMatchVisualInfo(display, screen, 32, 4, out var info);
+			var success = matchVisualInfoResult != 0;
+			if (!success)
+			{
+				var defaultDepth = XLib.XDefaultDepth(display, screen);
```

Comment:
I can not find the device that can enter this branch, so that I can not test this code in the real device.

Reply:
Again, we should assert and terminate on `!success`.

Reply:
I see you're already throwing on `!success` below, but after defaulting to `DefaultDepth` first. This will never fail indeed because there will always be a `Visual` with the default depth. Instead, let's try `24` and `32` and throw if neither are available.